### ISSUE TITLE
pcre2grep(1): Bug when context lines are consecutive

### ISSUE THREAD

**Issue by alejandro-colomar** [PCRE2Project/pcre2#577]
```sh
alx@devuan:~/tmp/grep$ cat foo 
zxc
cvb
bnm
asd
qwe
ert
tyu
uio
ggg
asd
dfg
ghj
jkl
123
345
567
789
alx@devuan:~/tmp/grep$ pcre2grep -nC3 '(ert|jkl)' foo
3-bnm
4-asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
--
10-asd
11-dfg
12-ghj
13:jkl
14-123
15-345
16-567
alx@devuan:~/tmp/grep$ pcre2grep -nC3 '(ert|ghj)' foo
3-bnm
4-asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
--
10-asd
11-dfg
12:ghj
13-jkl
14-123
15-345
alx@devuan:~/tmp/grep$ pcre2grep -nC3 '(ert|dfg)' foo
3-bnm
4-asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
--
10-asd
11:dfg
12-ghj
13-jkl
14-123
alx@devuan:~/tmp/grep$ pcre2grep -nC3 '(ert|asd)' foo
1-zxc
2-cvb
3-bnm
4:asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
10:asd
11-dfg
12-ghj
13-jkl
```

These lines should be printed without a separator, since they are consecutive.

Here's some reproducers with different -A and -B:

```sh
alx@devuan:~/tmp/grep$ pcre2grep -n -B3 -A3 '(ert|dfg)' foo
3-bnm
4-asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
--
10-asd
11:dfg
12-ghj
13-jkl
14-123
alx@devuan:~/tmp/grep$ pcre2grep -n -B2 -A4 '(ert|dfg)' foo
4-asd
5-qwe
6:ert
7-tyu
8-uio
9-ggg
10-asd
11:dfg
12-ghj
13-jkl
14-123
15-345
alx@devuan:~/tmp/grep$ pcre2grep -n -B4 -A2 '(ert|dfg)' foo
2-cvb
3-bnm
4-asd
5-qwe
6:ert
7-tyu
8-uio
--
9-ggg
10-asd
11:dfg
12-ghj
13-jkl
```

When a fix is applied, please include:
```
Reported-by: Alejandro Colomar <alx@kernel.org>
```

**Comment by PhilipHazel** [comment]
I can reproduce this; it does appear to be a bug and the output differs from GNU grep (which it shouldn't). Thanks for the report.

**Comment by PhilipHazel** [comment]
Fixed by f34fc0a3,

**Comment by alejandro-colomar** [comment]
Thank you!
